### ISSUE TITLE
In pycharm there is also no connect func for Signal

### ISSUE THREAD

**Issue by Adsicmes** [python-qt-tools/PySide6-stubs#1]
![image](https://github.com/python-qt-tools/PySide6-stubs/assets/79645438/d8b3f5c0-0267-4948-8de7-fd57465b1aa9)
![image](https://github.com/python-qt-tools/PySide6-stubs/assets/79645438/cc134f18-a757-48d8-bc79-60d875b39fe2)
![image](https://github.com/python-qt-tools/PySide6-stubs/assets/79645438/ceb1a7d6-79c0-47da-ac55-130295921be8)

As the image, connect func is in SignalInstance but not Signal

**Comment by altendky** [comment]
Have you tried `typing.ClassVar[PySide6.QtCore.Signal]`?  Without that, hint checkers won't know to use the `.__get__()` method (since Python would, in fact, not use it on an instance attribute either).

**Comment by Adsicmes** [comment]
The second and the third images are the code in your repo. I may not modify them.

**Comment by altendky** [comment]
Yes, I meant you should use `ClassVar` in your code.  Unbound signals are class attributes, not instance attributes, and are descriptors (https://docs.python.org/3/reference/datamodel.html#implementing-descriptors).  The `.__get__()` method is only used when accessing class attributes.  You have hinted your signals as instance attributes so the `.__get__()` won't be considered.

Also, shouldn't you be assigning a signal to that class attribute?  https://www.riverbankcomputing.com/static/Docs/PyQt6/signals_slots.html#defining-new-signals-with-pyqtsignal

**Comment by altendky** [comment]
I was, as you pointed out...  misreading your report.  I apologize for this.  I started https://github.com/python-qt-tools/PySide6-stubs/pull/4 and will ping Philippe.

**Comment by bluebird75** [comment]
Issue has been fixed, PyCharm now proposes connect functions in its completion
